extensions/2.0/Khronos/KHR_materials_volume/schema/glTF.KHR_materials_volume.schema.json

```diff
@@ -0,0 +1,37 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "title": "KHR_materials_volume glTF extension",
+    "type": "object",
+    "description": "glTF extension that defines the parameters for the volume of a material.",
+    "allOf": [ { "$ref": "glTFProperty.schema.json" } ],
+    "properties": {
+        "thicknessFactor": {
+            "type": "number",
+            "minimum": 0.0,
+            "default": 0.0,
+            "description": "Thickness of the volume in meters.",
+            "gltf_detailedDescription": "The thickness of the volume beneath the surface in meters. The value is relative to the local coordinate system of the mesh's node. A value greater than 0 turns the mesh into a volume with a homogeneous medium, enabling refraction, absorption and subsurface scattering. The actual value may be ignored by renderers that are able to derive the thickness from the mesh (ray tracer)."
```

Comment:
The first sentence here says "in meters", but in `attenuationDistance` it says "in world space".

Let's be consistent and use one term for both.  Which one is better, world space?  That way some other extension could redefine world distance units to be something other than meters.

Reply:
This conversation triggered a thought here.  In many applications, a scale transformation can be applied to models to fit into a scene.  What does a scaled model mean for the volume material? I think we might want to say that thickness and attenuation distance is in model space, implying that these values should also be scaled if the model is also scaled.

Reply:
@abwood wouldn't that be what "in world space" means? So when you have the same object in different sizes it still looks identical. Whereas "in meter" would mean that the objects change appearance with size. Two different behaviors, not sure which one to prefer.

Reply:
ah, didn't read carefully. You're saying this would be "model space". So "meter" and "world space" would be identical.

Reply:
Thinking this through further:  Imagine the case of a large wax candle with SSS that allows a bit of light to come through around the edges.  If a copy of this candle model gets scaled down and placed on a birthday cake, the smaller candle should allow light to scatter through the center of the candle in ways that the full-scale model did not allow.

But now consider the difference between "thickness" (from a map) and "attenuationDistance."  When scaling the model node up or down, thickness scales along with the node.  The material's attenuation distance does not scale.

Not sure if there's a perfect solution, but we should consider these effects.

Reply:
> But now consider the difference between "thickness" (from a map) and "attenuationDistance." When scaling the model node up or down, thickness scales along with the node. The material's attenuation distance does not scale.

This is how it should behave, we may have to explain it more clearly. As with any other map in the material (2D texture), the "thickness" scales with the node. The "attenuationDistance", on the other hand, is difficult to scale, because the scale might be non-uniform, and this is hard to handle in the renderer (ray length derived from attenuation distance suddenly becomes dependent on the direction due to the non-uniform scaling matrix). Apart from that, its hard to use: when assigning materials after placing objects in the scene (scale, rotate, position), it is hard to reason about the "local size" of objects (it's hidden behind potentially nested transformation matrices, shown in some hidden dialog boxes).

A quick check in Blender Cycles confirms that. This is a screenshot of Blender with two spheres of different size, but the same material sharing the density value (≈ attenuation distance). Both look the same.

![](https://user-images.githubusercontent.com/2821679/104808936-c2ac4b80-57e9-11eb-89a7-e128e11adf8c.png)

Reply:
To disassociate this with the issue of what units are used in the glTF file itself, can we avoid saying "meter" and use "world space" or "node space" instead?  I suspect that AttenuationDistance happens in "world space" but the thicknessFactor is actually "node space", because thickness scales with the node.  Would you agree?

Reply:
Changes added with f616d7b